nuxeo-core/nuxeo-core-event/src/main/java/org/nuxeo/ecm/core/work/WorkComputation.java

```diff
@@ -95,6 +95,7 @@ public void processRecord(ComputationContext context, String inputStreamName, Re
                         context.askForCheckpoint();
                         return;
                     }
+                    work.setWorkInstanceState(Work.State.RUNNING);
```

Comment:
Shouldn't it be outside of `storeState` configuration?

Reply:
The default WM changes the state to RUNNING when the Work is executed, here I do the same in the SWM.
The goal is not to store the Work state for sharing here, it is to indicate that the execution is done from a WM.
A Work executed by a WM uses the DLQ fallback, a Work executed by the recovery procedure will have a different state and will not use the DLQ (to avoid infinite loop).

Reply:
you are right, this should be outside of the storeState, was ok because unit test are run with the storeState option